### DISCUSSION TITLE
Power cycle simulation

### DISCUSSION THREAD

**Discussion by manjunathnilugal** [pybamm-team/PyBaMM#1800] [Q&A]
Dear Sir,
I am trying to run a power interpolant drive cycle simulation for 8760 data points. The battery is getting overcharged at particular point and simulation stops. Is there any possibility to overcome this or operate battery between particular SOC i.e 80% to 30%.
Battery size = 170 series 500 parallel,
Chen2020 Li-ion battery properties.
The battery should work between 80 -20%, the extra power is considered as dump power.
For reference I can share my .csv data file.
If there any suggestions to overcome this problem kindly let me know.
[power_ba.csv](https://github.com/pybamm-team/PyBaMM/files/7517788/power_ba.csv)

Regards, 
Manjunath

**Comment by brosaplanella** [comment]
You can find an example of how to bring custom drive cycles [in this script](https://github.com/pybamm-team/PyBaMM/blob/develop/examples/scripts/experiment_drive_cycle.py). For the particular battery of 170 cells in series and 500 in parallel, I think you will need to use our sister project [liionpack](https://www.github.com/pybamm-team/liionpack). Alternatively, you can rescale the drive cycle so it works with a single cell. 

For the overcharge, not sure how you can prevent it. You can define the initial state of charge in a simulation by passing an extra argument `initial_soc` to the `solve` method.

PS: I am transferring this thread to "Discussions" as I think fits better there.
